src/test/java/io/sniffy/testng/SocketTestNg.java

```diff
@@ -0,0 +1,47 @@
+package io.sniffy.testng;
+
+import io.sniffy.test.Count;
+import io.sniffy.test.SniffyAssertionError;
+import io.sniffy.socket.BaseSocketTest;
+import io.sniffy.socket.SocketExpectation;
+import org.testng.annotations.*;
+
+import java.net.UnknownHostException;
+
+@Listeners({QueryCounter.class,MustFailListener.class})
+public class SocketTestNg extends BaseSocketTest {
+
+    @BeforeSuite
+    public void beforeSuite() throws UnknownHostException {
+        BaseSocketTest.resolveLocalhost();
+    }
+
+    @BeforeTest
+    public void beforeTest() throws Throwable {
+        echoServerRule.before();
+    }
+
+    @AfterTest
+    public void afterTest() throws Throwable {
+        echoServerRule.after();
+    }
+
+
+    @Test
+    @SocketExpectation(connections = @Count(2))
+    public void testAllowedOneQuery() {
+        performSocketOperation();
+        performSocketOperation();
+    }
+
+    @Test(expectedExceptions = SniffyAssertionError.class)
+    @SocketExpectation(connections = @Count(2))
+    @MustFail(SniffyAssertionError.class)
+    public void testAllowedOneQuery_Fails() {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method names should not contain underscores](https://www.codacy.com/app/sniffy/sniffy/file/2496725563/issues/source?bid=3005960&fileBranchId=3034483#l40)